src/contracts/AccessListFactory.ts

```diff
@@ -0,0 +1,130 @@
+import { BigNumber, Signer } from 'ethers'
+import { Config } from '../config'
+import AccessListFactory from '@oceanprotocol/contracts/artifacts/contracts/accesslists/AccessListFactory.sol/AccessListFactory.json'
+import { generateDtName, sendTx, getEventFromTx, ZERO_ADDRESS } from '../utils'
+import { AbiItem, AccessListData, ReceiptOrEstimate } from '../@types'
+import { SmartContractWithAddress } from './SmartContractWithAddress'
+
+/**
+ * Provides an interface for Access List Factory contract
+ */
+export class AccesslistFactory extends SmartContractWithAddress {
+  getDefaultAbi() {
+    return AccessListFactory.abi as AbiItem[]
+  }
+
+  /**
+   * Instantiate AccessListFactory class
+   * @param {string} address The factory contract address.
+   * @param {Signer} signer The signer object.
+   * @param {string | number} [network] Network id or name
+   * @param {Config} [config] The configuration object.
+   * @param {AbiItem[]} [abi] ABI array of the smart contract
+   */
+  constructor(
+    address: string,
+    signer: Signer,
+    network?: string | number,
+    config?: Config,
+    abi?: AbiItem[]
+  ) {
+    super(address, signer, network, config, abi)
+    this.abi = abi || this.getDefaultAbi()
+  }
+
+  /**
+   * Create new Access List Contract
+   * @param {AccessListData} listData The data needed to create an NFT.
+   * @param {Boolean} [estimateGas] if True, return gas estimate
+   * @return {Promise<string|BigNumber>} The transaction hash or the gas estimate.
+   */
+  public async deployAccessListContract<G extends boolean = false>(
+    listData: AccessListData,
```

Comment:
I would advice you to delete that typing (AccessListData) and use function params

Reply:
ok sure.